news/_posts/2020-02-04-the-new-prestashop-test-framework.md

```diff
@@ -0,0 +1,96 @@
+
+On the other hand, you create your test logic by using these objects. The goal is to have really small test steps which then use the methods from the objects. This way, your tests don’t know what’s happening behind the scenes. They only call methods which act as an abstraction layer.
+
+This is a good illustration of the separation of concerns: when a page is updated, you only change the corresponding object in its separate file so that selectors and methods still work the same. When your test needs an update, you only change the logic in the steps.
+
+
+### Re-focus the test effort
+
+The third point was about the campaign itself. We wanted to redirect the campaign to be more effective, faster (the full Selenium campaign could last for up to 20 hours !) and easier to maintain. That would mean a shift in the focus of the campaign: instead of making a “do-it-all” campaign, we would split up the effort in 3 campaigns: 
+
+- A sanity tests campaign
+- An integration tests campaign
+- An end-to-end tests campaign
+
+Each of these campaigns had its perks.
+
+The **sanity campaign** is here to make sure devs don’t break some core functionalities of PrestaShop. Ran for every PR, it checks all the critical features of the app: the frontend catalog, the cart, the checkout process; but also the backend with the product section and the order section. It’s a fast campaign (about 8 min) and it must pass for the PR to be reviewed: a fail in this section means being back to the coding board for the incriminated dev, with all the shame it brings (not really, but we QA engineers like to think it does).
+
+The **integration campaign** is the big pièce de résistance: every feature of every page is tested one by one (with of course a few exceptions when it’s too complicated to automate). The goal here is to verify that all features (taken individually) still work as expected. That means testing tables, filters, and CRUD functionalities (Create, Read, Update, Delete).
+This campaign is the biggest so far, and my team and I are proud to say that it works wonderfully. There are still some errors sometimes but we hope to iron things out with time. It already proved itself useful by finding bugs ! It’s also very consistent: when it fails it’s always in the same spot and for the same reason.
+
+The **end-to-end campaign** will cover what’s left: it is used to test real user paths. The goal is to make sure all the actions a merchant can do are covered by tests, so that the Tech team can be confident about the level of quality of the application. The Product team (and the QA team) already have a good idea of those user paths (gathered through the interviews with merchants) and we will try to implement paths that are the most relevant to a merchant daily and weekly recurring actions.
```

Comment:
```suggestion
The **end-to-end campaign** will cover what’s left: it is used to test real user journeys. The goal is to make sure all the actions a merchant can do are covered by tests, so that the Tech team can be confident about the level of quality of the application. The Product team (and the QA team) already have a good idea of these user journeys (gathered through the interviews with merchants) and we will try to implement journeys that are the most relevant to a merchant daily and weekly recurring actions.
```